packages/lib-core/src/types/store.ts

```diff
@@ -81,4 +81,15 @@ export type PluginManager = {
    * Enabling a plugin puts all of its extensions into use. Disabling it does the opposite.
    */
   setPluginsEnabled: (config: { pluginName: string; enabled: boolean }[]) => void;
+
+  /**
+   * Update extensions which are currently in use.
+   *
+   * This function should be called whenever a change is detected in any feature flags that
+   * may be used by plugins to gate their extensions, including:
+   * - flags managed by the host application
+   * - flags contributed by plugins
+   * - flags managed by external services like Unleash
+   */
+  updateExtensions: () => void;
```

Comment:
Are there other reasons to call `updateExtensions` instead of flags? 
Keeping the name generic is probably a good call. But if it's only for flags we can be more specific.

Reply:
We originally had this function as `featureFlagChanged` and then we renamed it to `updateExtensions` as we're effectively triggering an update of extensions when the flags change. Since feature flag change is the only use case to trigger these updates (at least for now), maybe we could rename it to something like `handleFeatureFlagUpdate` if that makes it clearer? What do you think @cvogt @vojtechszocs ?

Reply:
We called it `updateExtensions` because it effectively forces the `PluginStore` to recompute the list of extensions which are currently in use.

We avoided adding a method like `setFeatureFlags: (flags: { [name: string]: boolean }) => void` since this would force the host application to store all flags in a single object that would be passed to this method. Instead, the `PluginStore` will use the `isFeatureFlagEnabled: (flag: string) => boolean` option to determine the value of the given flag when needed.

Reply:
I'd keep it as `updateExtensions` - it's more generic and future proof :smiley: but yeah, right now, a change in feature flags is basically the only thing that should trigger this function.

Reply:
I'm good with this.

Reply:
FYI @vidyanambiar will be following up on this PR based on our discussion here:  https://coreos.slack.com/archives/C02EQERSV3N/p1653499298943189 (issue https://issues.redhat.com/browse/HAC-1406). For now adding my LGTM